src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import duke.exception.DukeException;
+import duke.exception.DukeFileNotFoundException;
+import duke.exception.DukeCorruptedFileException;
+import duke.exception.DukeReadFileException;
+import duke.exception.DukeUnrecognisedTaskTypeException;
+import duke.exception.DukeWriteFileException;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
+import duke.utils.TaskList;
+
+public class Storage {
+    public static final String DEFAULT_FILE_PATH = "data/data.txt";
+
+    private String filePath;
+
+    public Storage() {
+        this.filePath = Storage.DEFAULT_FILE_PATH;
+    }
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
```

Comment:
I like how you have 2 different constructors to cater for the default file path when no location is specified.